FilmFestivalLoader/IDFA/parse_idfa_html.py

```diff
@@ -773,515 +437,598 @@ class FilmsParseState(Enum):
+                stack.change(state.AWAITING_FILM_TITLE)
+            case [state.AWAITING_FILM_TITLE, 'img', a] if len(a) > 2 and a[0][0] == 'alt':
+                self.film_title = a[0][1]
+                self._read_film()
+                stack.change(state.AWAITING_FILM_URL)
+            case [state.AWAITING_FILM_URL, 'footer', _]:
+                stack.change(state.DONE)
 
     def handle_endtag(self, tag):
         HtmlPageParser.handle_endtag(self, tag)
 
-        if self.state_stack.state_is(self.FilmsParseState.IN_FILM_PROPERTY) and tag == 'div':
-            self.state_stack.change(self.FilmsParseState.IN_FILM_PROPERTIES)
-        elif self.state_stack.state_is(self.FilmsParseState.IN_FILM_PROPERTIES) and tag == 'div':
+        stack = self.state_stack
+        state = self.FilmsParseState
+        match [stack.state(), tag]:
+            case [state.IN_TITLE, 'title']:
+                stack.change(state.AWAITING_DESCR)
+
+        if stack.state_is(state.IN_FILM_PROPERTY) and tag == 'div':
+            stack.change(state.IN_FILM_PROPERTIES)
+        elif stack.state_is(state.IN_FILM_PROPERTIES) and tag == 'div':
             self.reset_film_parsing()
 
     def handle_data(self, data):
         HtmlPageParser.handle_data(self, data)
 
-        if self.state_stack.state_is(self.FilmsParseState.IN_DESCR):
-            if not data.startswith('.css'):
-                section_description = data
-                self.subsection = self.get_subsection(section_description)
-                self.state_stack.change(self.FilmsParseState.AWAITING_FILM_URL)
-            else:
-                COUNTER.increase('sections with css data')
-        elif self.state_stack.state_is(self.FilmsParseState.IN_FILM_TITLE):
-            self.film_title = data
-            COUNTER.increase('film title')
-            self.state_stack.change(self.FilmsParseState.IN_FILM_PROPERTIES)
-        elif self.state_stack.state_is(self.FilmsParseState.IN_FILM_PROPERTY):
-            self.film_property_by_label[self.metadata_key] = data
+        stack = self.state_stack
+        state = self.FilmsParseState
+        match stack.state():
+            case state.IN_TITLE:
+                self.section_title = data.split('|')[-1].strip()
+
+    def _read_film(self):
+        comment(f'About to parse film {self.film_title} from {self.film_url}')
+        get_film_from_theme_part_page(self.festival_data, self.film_title, self.film_url, self.pathway_url)
+        self._init_film_data()
+
+    def _init_film_data(self):
+        self.film_url = None
+        self.film_title = None
+        self.section_title = None
+        self.section_name = None
+        self.subsection = None
+        self.film_property_by_label = {}
+        self.metadata_key = None
+
+    def reset_film_parsing(self):
+        DEBUG_RECORDER.add('RESET FILM PARSING, is pop() OK?')
+        self.state_stack.pop()
+        self._init_film_data()
+
```

Comment:
Is that debug statement still needed?

Reply:
Changed the debug statement with question in a stack dump.
No `pop()` anymore, stack also dumped on the console.